### ISSUE TITLE
Code generation hard wraps prematurely after a forEach lambda

### ISSUE THREAD

**Issue by SamuelFrankSmith** [apollographql/apollo-android#2620]
**Summary**
Auto-generated Input data class hard wraps at 100 characters immediately after a `forEach` lambda, but before the lambda opening brace, `{`. 
This causes a compilation error, specifically `Function invocation 'forEach(...)' expected`

**Version**
2.4.0

**Description**
Example of what was occuring during compilation:
```
    writer.writeList("contraParticipantEntries") { listItemWriter ->
      this@ObfuscatedObjectNameButStillIllustratesTheIssueABCDInput.contraParticipantEntries.forEach
          { value ->
        listItemWriter.writeObject(value.marshaller())
      }
    }
    writer.writeInt("newEntryInsertionIndex",
        this@ObfuscatedObjectNameButStillIllustratesTheIssueABCDInput.newEntryInsertionIndex)
  }
```
Full error list at compilation:
```
Function invocation 'forEach(...)' expected
No value passed for parameter 'action'
Cannot infer a type for this parameter. Please specify it explicitly.
```
I thought it was an odd issue, so I updated our `schema.json` to shorten the input's name from `ObfuscatedObjectNameButStillIllustratesTheIssueABCDInput` to 
`ObjectNameButStillIllustratesTheIssueABCInput` and reran the compilation. 
The result was successful, and the above code correctly wrapped after the parameter: `.forEach { value ->`
_Further inspection indicated that the opening brace needs to be on the same line as the function call, but not the parameter._

_I assumed that the auto-generated code was wrapped by my editor's hard wrap length, but that does not appear to be the case, and now I'm writing this ticket 👍_

Edit: Might be related to #1622, #1296

**Comment by martinbonnin** [comment]
Thanks for the detailed bug report ! Indeed, it looks like we need to add some of these [non-wrapping magical characters](https://square.github.io/kotlinpoet/#spaces-wrap-by-default). 

Do you feel like opening a PR? Else I can look into it in the coming days.

**Comment by martinbonnin** [comment]
Closed by #2621
